include/git2/patch.h

```diff
@@ -267,6 +267,35 @@ GIT_EXTERN(int) git_patch_to_buf(
 	git_buf *out,
 	git_patch *patch);
 
+/**
+ * Apply a patch to a buffer.
+ *
+ * Attempts to apply the changes represented by a patch to a source buffer
+ * producing a new output buffer. A hunk callback can be given to choose
+ * which hunks are applied.
+ *
+ * @param out The output buffer to be filled
+ * @param filename The output name of the new file
+ * @param mode The output mode of the new file
```

Comment:
Do you mean to make all of the out parameters optional or just the file path and mode? I suppose that if the user only cares about the result code (whether the patch can be applied cleanly) then it would make sense for the output buffer to be optional too.